BlackNectar/BlackNectar/StoresTableViewController.swift

```diff
@@ -128,41 +132,66 @@ extension StoresTableViewController {
     
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         
-        return stores.count
+        return stores.notEmpty ? stores.count : 1
         
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: "storeCell", for: indexPath) as? StoresTableViewCell else {
+        if stores.notEmpty {
             
-            LOG.error("Failed to dequeue StoresTableViewCell")
-            return UITableViewCell()
+            tableView.rowHeight = 236
+            
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "storeCell", for: indexPath) as? StoresTableViewCell else {
+                
+                LOG.error("Failed to dequeue StoresTableViewCell")
+                return UITableViewCell()
+                
+            }
+            
+            let row = indexPath.row
+            
+            guard stores.isInBounds(index: row) else {
+                LOG.warn("Received Out of Bounds Index: \(row)")
+                return cell
+            }
+            
+            let store = stores[row]
+            
+            insertDistance(toStore: store, into: cell)
+            goLoadImage(into: cell, withStore: store.storeImage)
+            insertAddress(into: cell, withStore: store)
+            
+            cell.storeName.text = store.storeName
+            cell.onGoButtonPressed = { cell in
```

Comment:
Make sure you use `[weak cell]` here, so that the closure doesn't keep the cell alive.

Read about retain cycles here:
https://krakendev.io/blog/weak-and-unowned-references-in-swift